### PR TITLE
Update stml475 gsg model to v2

### PR DESCRIPTION
**Pull request by ryanwinter**
## Purpose
Update the GSG to v2 of the dtdl model to support Central

## Does this introduce a breaking change?
```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone https://github.com/ryanwinter/iot-middleware-freertos-samples.git
```

* Test the code
```
https://github.com/ryanwinter/iot-middleware-freertos-samples/blob/main/demos/projects/ST/b-l475e-iot01a/GSG.md
```

## What to Check
Verify that the following are valid
* Guide completes successfully

### CODE REVIEW

**Review by wduraes — APPROVED** [review]

File: demos/projects/ST/b-l475e-iot01a/sample_gsg_device.c [hunk 1]
```diff
@@ -36,7 +36,7 @@
 #define TELEMETRY_GYROSCOPEY        ( "gyroscopeY" )
 #define TELEMETRY_GYROSCOPEZ        ( "gyroscopeZ" )
 
-const char * pcModelId = "dtmi:azureiot:devkit:freertos:gsgstml475;1";
+const char * pcModelId = "dtmi:azureiot:devkit:freertos:gsgstml475;2";
 
 const char * pcManufacturerPropertyValue = "STMicroelectronics";
```
Comment (hihigupt) [inline demos/projects/ST/b-l475e-iot01a/sample_gsg_device.c:41]: >pcManufacturerPropertyValue [](http://example.com/codeflow?start=13&length=27)

should we keep strings as const char or macros

**Comment by hihigupt** [comment]
> const double pxTotalStoragePropertyValue = 8192;

this is not pointer ... please follow this link for code style.

const double xTotalStoragePropertyValue = 8192;

[https://www.freertos.org/FreeRTOS-Coding-Standard-and-Style-Guide.html](https://www.freertos.org/FreeRTOS-Coding-Standard-and-Style-Guide.html)

---
Refers to: demos/projects/ST/b-l475e-iot01a/sample_gsg_device.c:47 in 60f3668. [](commit_id = 60f366821945986f90df553ed89ba30fe2c1e6a9, deletion_comment = False)

**Comment by hihigupt** [comment]
> static void createTelemetryDevice( AzureIoTJSONWriter_t * xWriter )

name of function is formated following way:

returntypeFunctionName()

so, for this function it will be:
static void vCreateTelemetryDevice( AzureIoTJSONWriter_t * pxWriter );

---
Refers to: demos/projects/ST/b-l475e-iot01a/sample_gsg_device.c:62 in 60f3668. [](commit_id = 60f366821945986f90df553ed89ba30fe2c1e6a9, deletion_comment = False)

**Comment by hihigupt** [comment]
>     BSP_MAGNETO_GetXYZ( magnetometer );

Please use naming of variables based on the freertos guide, as for this repo we are trying to follow freertos guidelines.

[https://www.freertos.org/FreeRTOS-Coding-Standard-and-Style-Guide.html](https://www.freertos.org/FreeRTOS-Coding-Standard-and-Style-Guide.html)

---
Refers to: demos/projects/ST/b-l475e-iot01a/sample_gsg_device.c:85 in 60f3668. [](commit_id = 60f366821945986f90df553ed89ba30fe2c1e6a9, deletion_comment = False)

**Comment by hihigupt** [comment]
>     return lBytesWritten;

should we cast this return variable ?

---
Refers to: demos/projects/ST/b-l475e-iot01a/sample_gsg_device.c:199 in 60f3668. [](commit_id = 60f366821945986f90df553ed89ba30fe2c1e6a9, deletion_comment = False)

**Comment by hihigupt** [comment]
>     return lBytesWritten;

Can we also make check in CI test that build system produces this binary..

[https://github.com/Azure-Samples/iot-middleware-freertos-samples/blob/79200b22df0dd93fe3aab1159d436fb9e9efa4bd/.github/scripts/ci_tests.sh#L48](https://github.com/Azure-Samples/iot-middleware-freertos-samples/blob/79200b22df0dd93fe3aab1159d436fb9e9efa4bd/.github/scripts/ci_tests.sh#L48)

exit_if_binary_does_not_exist "build_st_b-l475e-iot01a" "iot-middleware-sample-gsg.elf"


---
Refers to: demos/projects/ST/b-l475e-iot01a/sample_gsg_device.c:199 in 60f3668. [](commit_id = 60f366821945986f90df553ed89ba30fe2c1e6a9, deletion_comment = False)

**Comment by hihigupt** [comment]
>     return lBytesWritten;

this need casting

---
In reply to: [899691378](https://github.com/Azure-Samples/iot-middleware-freertos-samples/pull/78#issuecomment-899691378) [](http://example.com/codeflow?ancestors=899691378)

---
Refers to: demos/projects/ST/b-l475e-iot01a/sample_gsg_device.c:199 in 60f3668. [](commit_id = 60f366821945986f90df553ed89ba30fe2c1e6a9, deletion_comment = False)

File: demos/projects/ST/b-l475e-iot01a/sample_gsg_device.c (hunk 1 above)
Comment (ryanwinter) [reply demos/projects/ST/b-l475e-iot01a/sample_gsg_device.c:41]: Currently they are stored as strings, as they linker is used to link in the correct version. We could move to using the preprocessor, would possibly require additional #ifdefs and such which would look a little messier

**Comment by ryanwinter** [comment]
> > const double pxTotalStoragePropertyValue = 8192;
> 
> this is not pointer ... please follow this link for code style.
> 
> const double xTotalStoragePropertyValue = 8192;
> 
> https://www.freertos.org/FreeRTOS-Coding-Standard-and-Style-Guide.html
> 
> Refers to: demos/projects/ST/b-l475e-iot01a/sample_gsg_device.c:47 in [60f3668](https://github.com/Azure-Samples/iot-middleware-freertos-samples/commit/60f366821945986f90df553ed89ba30fe2c1e6a9). [](commit_id = [60f3668](https://github.com/Azure-Samples/iot-middleware-freertos-samples/commit/60f366821945986f90df553ed89ba30fe2c1e6a9), deletion_comment = False)

fixed

**Comment by hihigupt** [comment]
> 

this check will make "t", "tr" and "tru" prefixes return true.

---
Refers to: demos/sample_azure_iot_gsg/sample_azure_iot_gsg.c:329 in 60f3668. [](commit_id = 60f366821945986f90df553ed89ba30fe2c1e6a9, deletion_comment = False)

**Comment by hihigupt** [comment]
>                 int32_t lNewTelemetryInterval,

can we keep one line per defination ? Also please keep the defenation at begining of function.

---
Refers to: demos/sample_azure_iot_gsg/sample_azure_iot_gsg.c:430 in 60f3668. [](commit_id = 60f366821945986f90df553ed89ba30fe2c1e6a9, deletion_comment = False)

**Comment by ryanwinter** [comment]
> > static void createTelemetryDevice( AzureIoTJSONWriter_t * xWriter )
> 
> name of function is formated following way:
> 
> returntypeFunctionName()
> 
> so, for this function it will be:
> static void vCreateTelemetryDevice( AzureIoTJSONWriter_t * pxWriter );
> 
> Refers to: demos/projects/ST/b-l475e-iot01a/sample_gsg_device.c:62 in [60f3668](https://github.com/Azure-Samples/iot-middleware-freertos-samples/commit/60f366821945986f90df553ed89ba30fe2c1e6a9). [](commit_id = [60f3668](https://github.com/Azure-Samples/iot-middleware-freertos-samples/commit/60f366821945986f90df553ed89ba30fe2c1e6a9), deletion_comment = False)

resolved all functions names in the device gsg files

**Comment by hihigupt** [comment]
>                 if( xResult != eAzureIoTSuccess )

can we stick to either doing configAssert for parsing error or log error.

---
Refers to: demos/sample_azure_iot_gsg/sample_azure_iot_gsg.c:433 in 60f3668. [](commit_id = 60f366821945986f90df553ed89ba30fe2c1e6a9, deletion_comment = False)

**Comment by hihigupt** [comment]
> #include "sample_gsg_device.h"

General include sequence followed is:


In dir/foo.cc or dir/foo_test.cc, whose main purpose is to implement or test the stuff in dir2/foo2.h, order your includes as follows:

dir2/foo2.h.
A blank line
C system headers (more precisely: headers in angle brackets with the .h extension), e.g., <unistd.h>, <stdlib.h>.
A blank line
C++ standard library headers (without file extension), e.g., <algorithm>, <cstddef>.
A blank line
Other libraries' .h files.
Your project's .h files.

---
Refers to: demos/sample_azure_iot_gsg/sample_azure_iot_gsg.c:4 in 60f3668. [](commit_id = 60f366821945986f90df553ed89ba30fe2c1e6a9, deletion_comment = False)

**Comment by hihigupt** [comment]
> > > static void createTelemetryDevice( AzureIoTJSONWriter_t * xWriter )
> > 
> > 
> > name of function is formated following way:
> > returntypeFunctionName()
> > so, for this function it will be:
> > static void vCreateTelemetryDevice( AzureIoTJSONWriter_t * pxWriter );
> > Refers to: demos/projects/ST/b-l475e-iot01a/sample_gsg_device.c:62 in [60f3668](https://github.com/Azure-Samples/iot-middleware-freertos-samples/commit/60f366821945986f90df553ed89ba30fe2c1e6a9). [](commit_id = [60f3668](https://github.com/Azure-Samples/iot-middleware-freertos-samples/commit/60f366821945986f90df553ed89ba30fe2c1e6a9), deletion_comment = False)
> 
> resolved all functions names in the device gsg files

Just make sure static functions are also named prv prefixed

**Comment by ryanwinter** [comment]
> > ```
> > BSP_MAGNETO_GetXYZ( magnetometer );
> > ```
> 
> Please use naming of variables based on the freertos guide, as for this repo we are trying to follow freertos guidelines.
> 
> https://www.freertos.org/FreeRTOS-Coding-Standard-and-Style-Guide.html
> 
> Refers to: demos/projects/ST/b-l475e-iot01a/sample_gsg_device.c:85 in [60f3668](https://github.com/Azure-Samples/iot-middleware-freertos-samples/commit/60f366821945986f90df553ed89ba30fe2c1e6a9). [](commit_id = [60f3668](https://github.com/Azure-Samples/iot-middleware-freertos-samples/commit/60f366821945986f90df553ed89ba30fe2c1e6a9), deletion_comment = False)

update all variable names to follow coding style

**Comment by ryanwinter** [comment]
> > ```
> > return lBytesWritten;
> > ```
> 
> should we cast this return variable ?
> 
> Refers to: demos/projects/ST/b-l475e-iot01a/sample_gsg_device.c:199 in [60f3668](https://github.com/Azure-Samples/iot-middleware-freertos-samples/commit/60f366821945986f90df553ed89ba30fe2c1e6a9). [](commit_id = [60f3668](https://github.com/Azure-Samples/iot-middleware-freertos-samples/commit/60f366821945986f90df553ed89ba30fe2c1e6a9), deletion_comment = False)

done

**Comment by ryanwinter** [comment]
> > ```
> > return lBytesWritten;
> > ```
> 
> Can we also make check in CI test that build system produces this binary..
> 
> https://github.com/Azure-Samples/iot-middleware-freertos-samples/blob/79200b22df0dd93fe3aab1159d436fb9e9efa4bd/.github/scripts/ci_tests.sh#L48
> 
> exit_if_binary_does_not_exist "build_st_b-l475e-iot01a" "iot-middleware-sample-gsg.elf"
> 
> Refers to: demos/projects/ST/b-l475e-iot01a/sample_gsg_device.c:199 in [60f3668](https://github.com/Azure-Samples/iot-middleware-freertos-samples/commit/60f366821945986f90df553ed89ba30fe2c1e6a9). [](commit_id = [60f3668](https://github.com/Azure-Samples/iot-middleware-freertos-samples/commit/60f366821945986f90df553ed89ba30fe2c1e6a9), deletion_comment = False)

done

**Comment by ryanwinter** [comment]
> > 
> 
> this check will make "t", "tr" and "tru" prefixes return true.
> 
> Refers to: demos/sample_azure_iot_gsg/sample_azure_iot_gsg.c:329 in [60f3668](https://github.com/Azure-Samples/iot-middleware-freertos-samples/commit/60f366821945986f90df553ed89ba30fe2c1e6a9). [](commit_id = [60f3668](https://github.com/Azure-Samples/iot-middleware-freertos-samples/commit/60f366821945986f90df553ed89ba30fe2c1e6a9), deletion_comment = False)

fixed

**Comment by ryanwinter** [comment]
> > #include "sample_gsg_device.h"
> 
> General include sequence followed is:
> 
> In dir/foo.cc or dir/foo_test.cc, whose main purpose is to implement or test the stuff in dir2/foo2.h, order your includes as follows:
> 
> dir2/foo2.h.
> A blank line
> C system headers (more precisely: headers in angle brackets with the .h extension), e.g., <unistd.h>, <stdlib.h>.
> A blank line
> C++ standard library headers (without file extension), e.g., , .
> A blank line
> Other libraries' .h files.
> Your project's .h files.
> 
> Refers to: demos/sample_azure_iot_gsg/sample_azure_iot_gsg.c:4 in [60f3668](https://github.com/Azure-Samples/iot-middleware-freertos-samples/commit/60f366821945986f90df553ed89ba30fe2c1e6a9). [](commit_id = [60f3668](https://github.com/Azure-Samples/iot-middleware-freertos-samples/commit/60f366821945986f90df553ed89ba30fe2c1e6a9), deletion_comment = False)

fixed header includes in sample_azure_iot_gsg.c

**Review by hihigupt — APPROVED** [review]

**Comment by hihigupt** [comment]
Thanks @ryanwinter  for taking care of nitpicks...:)

**Comment by ryanwinter** [comment]
> Thanks @ryanwinter for taking care of nitpicks...:)

Actually all your feedback was super valuable!

**Pull request merged**